django/contrib/contenttypes/management.py

```diff
@@ -1,9 +1,65 @@
 from django.apps import apps
-from django.db import DEFAULT_DB_ALIAS, router
+from django.core.management.color import color_style
+from django.db import DEFAULT_DB_ALIAS, migrations, router, transaction
+from django.db.utils import IntegrityError
 from django.utils import six
 from django.utils.six.moves import input
 
 
+class RenameContentType(migrations.RunPython):
+    def __init__(self, app_label, old_name, new_name, verbosity):
+        self.app_label = app_label
+        self.old_name = old_name
+        self.new_name = new_name
+        self.verbosity = verbosity
+        super(RenameContentType, self).__init__(self.rename_forward, self.rename_backward)
+
+    def _rename(self, apps, old_name, new_name):
+        ContentType = apps.get_model('contenttypes', 'ContentType')
+        try:
+            content_type = ContentType.objects.get_by_natural_key(self.app_label, old_name.lower())
+        except ContentType.DoesNotExist:
+            pass
+        else:
+            content_type.model = new_name.lower()
+            using = router.db_for_write(ContentType, instance=content_type)
+            try:
+                with transaction.atomic(using=using):
+                    content_type.save(update_fields={'model'})
+            except IntegrityError:
+                # Gracefully fallback if a stale content type causes a conflict.
+                if self.verbosity:
+                    print(color_style().WARNING(
+                        "Failed to rename content type '%(app_label)s.%(old_name)s' "
+                        "to '%(app_label)s.%(new_name)s'.\nYou might have stale content types in conflict." % {
+                            'app_label': self.app_label,
+                            'old_name': old_name,
+                            'new_name': new_name,
+                        }
+                    ))
```

Comment:
We could insert an appropriate `CreateContentType` operation after `CreateModel` but it would have to deal with the same conflict issue.

In an ideal world no one would have stale content type, `ContentTypeManager` methods wouldn't use `get_or_create` (which I suspect was introduced because Django didn't handle migrations back then), `ContentType` would be created only by `CreateContentType` and `DeleteModel` would be followed by a `DeleteContentType` operation that automatically follows `on_delete` pointing at `ContentType`.